src/device/device.go

```diff
@@ -17,8 +23,49 @@ func (device *Device) String() string {
 	return "Device:{ deviceId:" + strconv.Itoa(device.ID) + ", Host:" + device.Host + ", port:" + device.Port + " }"
 }
 
+//Init - Initialize all the components on this device
+func (device *Device) Init() error {
+	for i := 0; i < len(device.Solenoids); i++ {
+		err := device.Solenoids[i].Init()
+		if err != nil {
+			log.Println("Failed to load: ", device.Solenoids[i].String())
+		}
+	}
+	return nil
+}
+
 //ToFullAddress returns a network address including the ip address and port that this device is listening on
 func (device *Device) ToFullAddress() string {
 	/* Just for pretty printing the device info */
 	return device.Host + ":" + device.Port
 }
+
+//LoadSolenoids - Load Solenoid Array from Config
+func (device *Device) LoadSolenoids() error {
+	yamlFile, err := ioutil.ReadFile("./app/config/solenoids.yaml")
+	if err != nil {
+		log.Printf("yamlFile.Get err   #%v ", err)
+		return err
+	}
+	err = yaml.Unmarshal(yamlFile, &device)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+		return err
+	}
+	log.Println("Loaded the Following Solenoids: ", device.solenoidsString())
+	log.Print("Performing Initializations: ")
+	err = device.Init()
+	if err != nil {
+		return err
```

Comment:
If you are going to log.Printf errors for the other error checks maybe we should put them here

Reply:
added missing print